File: glucare_java/glucare/src/main/java/com/uade/glucare/model/Medicine.java
package com.uade.glucare.model;

import jakarta.annotation.Generated;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "medicine")
public class Medicine {

    @Id
    @GeneratedValue
    @Column(name = "id")
    private Long id;

    @Column(name = "nombre")
    private String nombre;

    @Column(name = "dosis")
    private double dosis;

    @Column(name = "frecuencia")
    private String frecuencia;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    public Long getId() {
        return id;
    }

    public String getNombre() {
        return nombre;
    }

    public double getDosis() {
        return dosis;
    }

    public String getFrecuencia() {
        return frecuencia;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setDosis(double dosis) {
        this.dosis = dosis;
    }

    public void setFrecuencia(String frecuencia) {
        this.frecuencia = frecuencia;
    }
    
}

File: glucare_flutter/lib/model/Medicine.dart
import 'package:glucare/model/User.dart';

class Medicine {
  final int id;
  final String nombre;
  final double dosis;
  final String frecuencia;
  final User user;

  Medicine({
    required this.id,
    required this.nombre,
    required this.dosis,
    required this.frecuencia,
    required this.user,
  });

  factory Medicine.fromJson(Map<String, dynamic> json) {
    return Medicine(
      id: json['id'],
      nombre: json['nombre'],
      dosis: json['dosis'],
      frecuencia: json['frecuencia'],
      user: User.fromJson(json['user']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'nombre': nombre,
      'dosis': dosis,
      'frecuencia': frecuencia,
      'user': user.toJson(),
    };
  }
}